include/libsemigroups/fpsemi-examples.hpp

```diff
@@ -0,0 +1,539 @@
+//
+// libsemigroups - C++ library for semigroups and monoids
+// Copyright (C) 2019 Florent Hivert
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+#ifndef LIBSEMIGROUPS_FPSEMI_EXAMPLES_HPP_
+#define LIBSEMIGROUPS_FPSEMI_EXAMPLES_HPP_
+
+#include <cctype>         // for toupper
+#include <cstddef>        // for size_t
+#include <string>         // for basic_string, operator+
+#include <unordered_set>  // for unordered_set
+#include <vector>         // for vector
+
+#include "libsemigroups/present.hpp"  // for Presentation
+#include "libsemigroups/types.hpp"    // for relation_type
+
+namespace libsemigroups {
+  enum class author : uint64_t {
```

Comment:
This should be documented too, see for example, 

https://libsemigroups.readthedocs.io/en/latest/types.html